File: editor/en/[2073]Time Needed to Buy Tickets.java

//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
    public int timeRequiredToBuy(int[] tickets, int k) {
        int count = 0;

        for (int i = 0; i < tickets.length; i++) {
            if (tickets[i] >= benchMark) {
                if (i <= k) {
                    count += benchMark;
                } else {
                    count += benchMark - 1;
                }
            } else {
                count += tickets[i];
            }
        }

        return count;
    }
}
/*
class Solution {
    public int timeRequiredToBuy(int[] tickets, int k) {
        int count = 0;

        for (int i = 0; i < tickets.length; i++) {
            count += Math.min(tickets[i], tickets[k] - (i > k ? 1 : 0));
        }

        return count;
    }
}
 */
//leetcode submit region end(Prohibit modification and deletion)

File: editor/en/[2073]Time Needed to Buy Tickets.py
# leetcode submit region begin(Prohibit modification and deletion)
class Solution:
    def timeRequiredToBuy(self, tickets: List[int], k: int) -> int:
        bench_mark = tickets[k]
        count = 0

        for i in range(len(tickets)):
            if tickets[i] >= bench_mark:
                if i <= k:
                    count += bench_mark
                else:
                    count += bench_mark - 1
            else:
                count += tickets[i]

        return count

    # return sum(min(tickets[i], tickets[k] - (i > k)) for i in range(len(tickets)))
# leetcode submit region end(Prohibit modification and deletion)
